barretenberg/cpp/src/barretenberg/boomerang_value_detection/graph.cpp

```diff
@@ -0,0 +1,824 @@
+        for (size_t i = 0; i < arithmetic_gates_numbers; i++) {
+            auto gate_variables = get_arithmetic_gate_connected_component(ultra_circuit_constructor, i);
+            connect_all_variables_in_vector(ultra_circuit_constructor, gate_variables, false);
+        }
+    }
+    const auto& elliptic_block = ultra_circuit_constructor.blocks.elliptic;
+    auto elliptic_gates_numbers = elliptic_block.size();
+    bool elliptic_gates_exist = elliptic_gates_numbers > 0;
+    if (elliptic_gates_exist) {
+        for (size_t i = 0; i < elliptic_gates_numbers; i++) {
+            std::vector<uint32_t> gate_variables = get_elliptic_gate_connected_component(ultra_circuit_constructor, i);
+            connect_all_variables_in_vector(ultra_circuit_constructor, gate_variables, false);
+        }
+    }
+    const auto& range_block = ultra_circuit_constructor.blocks.delta_range;
+    auto range_gates = range_block.size();
+    bool range_gates_exists = range_gates > 0;
+    if (range_gates_exists) {
+        std::vector<uint32_t> sorted_variables;
+        for (size_t i = 0; i < range_gates; i++) {
+            auto current_gate = get_sort_constraint_connected_component(ultra_circuit_constructor, i);
+            if (!current_gate.empty()) {
+                sorted_variables.insert(sorted_variables.end(), current_gate.begin(), current_gate.end());
+            } else {
+                connect_all_variables_in_vector(ultra_circuit_constructor, sorted_variables, true);
+                sorted_variables.clear();
+            }
+        }
+    }
+
+    const auto& lookup_block = ultra_circuit_constructor.blocks.lookup;
+    auto lookup_gates = lookup_block.size();
+    bool lookup_gates_exists = lookup_gates > 0;
+    if (lookup_gates_exists) {
+        for (size_t i = 0; i < lookup_gates; i++) {
+            std::vector<uint32_t> variable_indices = get_plookup_gate_connected_component(ultra_circuit_constructor, i);
+            connect_all_variables_in_vector(ultra_circuit_constructor, variable_indices, false);
+        }
+    }
+}
+
+/**
+ * @brief this method checks that variable with given index is not constant
+ *
+ * @tparam FF
+ * @param ultra_circuit_builder
+ * @param variable_index
+ * @return true
+ * @return false
+ */
+
+template <typename FF>
+bool Graph_<FF>::check_is_not_constant_variable(bb::UltraCircuitBuilder& ultra_circuit_builder,
+                                                const uint32_t& variable_index)
+{
+    bool is_not_constant = true;
+    auto constant_variable_indices = ultra_circuit_builder.constant_variable_indices;
+    for (const auto& pair : constant_variable_indices) {
+        if (pair.second == ultra_circuit_builder.real_variable_index[variable_index]) {
+            is_not_constant = false;
+            break;
+        }
+    }
+    return is_not_constant;
+}
+
+/**
+ * @brief this method connects between each other variables that have different indexes, if their indexes are not equal
```

Comment:
Adds connections between them. But if I were you, I would rephrase the whole sentence.